Added 5.12.0 changelog (#1693)
sensu/sensu-docs: content/sensu-go/5.12/release-notes.md

---

## 5.12.0 release notes

**August 22, 2019** &mdash; The latest release of Sensu Go, version 5.12.0, is now available for download. There are some exciting feature additions in this release including the ability to output resources to a file from sensuctl and more granularity of control around checks/check hooks execution with an agent allow list. Additionally, this release includes the capability to delete assets and some more stability fixes around watcher functionality. See the [upgrading guide]() to upgrade Sensu to version 5.12.0.

**IMPORTANT:**

Due to changes in the release process, Sensu binary-only archives are now named following the pattern `sensu-go_5.12.0_$OS_$ARCH.tar.gz`, where $OS is the operating system name and $ARCH is the CPU architecture. These archives include all files in the top level directory. See the [installation guide][16] for the latest download links.
Line item

**NEW FEATURES:**

> Added sensu-agent and sensuctl binary builds for FreeBSD 

Reviewer: ```suggestion
Operators can now authenticate to Sensu via OpenID Direct Connect (OIDC) using sensuctl, see our [authentication documentation][17] for details.
Added sensu-agent and sensuctl binary builds for FreeBSD on amd64 and 386 architectures.
```

Reviewer: TODO: add a link for authentication documentation at the end of the doc.

> Added sensuctl dump command to output resources to a file or STDOUT

Reviewer: ```suggestion
Added sensuctl dump command to output resources to a file or STDOUT, making it easier to backup your Sensu backends.
```

> Added agent --allow-list to list allowed executables for checks and check hooks.

Reviewer: ```suggestion
Agents can now be configured with a list executables which check and check hook commands are allowed to run.
```

**IMPROVEMENTS:**

> Licensed feature: Namespaces listed in both the web UI and sensuctl are now limited to those which the user has access to

Reviewer: ```suggestion
Assets now support defining multiple builds, reducing the number of individual assets needed to cover disparate platforms in your infrastructure.
Licensed feature: Namespaces listed in both the web UI and sensuctl are now limited to those which the user has access to.
```

Licensed feature: Namespaces listed in both the web UI and sensuctl are now limited to those which the user has access to.
The event.check.name field has been added as a supported field selector

> Both the API and sensuctl can now be used to delete Assets

Reviewer: ```suggestion
Both the API and sensuctl can now be used to delete Assets.
```

> We fixed some bugs in watchers through refactoring -- what does it mean to end users?

Reviewer 2: Users who noticed checks failing to update or schedule should see an improvement.

> We fixed some bugs in watchers through refactoring -- what does it mean to end users?

Reviewer: We'll move this down to fixes. Suggest we replace it with something to cover Eric's comments about web UI speed improvements.

```suggestion
Web UI performance is improved UI for deployments with lots of events and entities.
```

> The generic etcd watcher now keeps track of revisions.

Reviewer 2: Probably best to not mention this.

Author: ```suggestion
```

> The generic etcd watcher now keeps track of revisions.

Reviewer: ```suggestion
```

> Web UI performance is improved UI for deployments with lots of events and entities.

Reviewer 2: The English is a bit English in this sentence :joy:

> Web UI performance is improved UI for deployments with lots of events and entities.

Reviewer: ```suggestion
Web UI performance is improved for deployments with many events and entities.
```

> The resource caches can now rebuild themselves in case of failures.

Reviewer: @echlebek is this a user-facing improvement?

Reviewer 2: Users with large inventories should notice an improvement.

Reviewer: @echlebek are these caches on the backend, agent or both?

Reviewer 2: They are on the backend. This is an implementation detail, it doesn't necessarily need to be communicated to users.

Event and Entity resources can now be created via the API without an explicit namespace; the system will refer to the namespace in the request URL.

> Event and Entity resources can now be created via the API using the POST verb

Reviewer: ```suggestion
Event and Entity resources can now be created via the API using the POST verb.
```

**SECURITY:**

To prevent writing sensitive data to logs, decoded check result and keepalive payloads are no longer logged by the backend.

**FIXES:**

Tabular display of filters via Sensuctl now displays && or || as appropriate for inclusive and exclusive filters, respectively.

> Requesting events from the GET /events/:entity API endpoint now returns events only for the specified entity

Reviewer: ```suggestion
Requesting events from the GET /events/:entity API endpoint now returns events only for the specified entity.
```

> Running sensuctl config view without configuration no longer causes a crash

Reviewer: ```suggestion
Running sensuctl config view without configuration no longer causes a crash.
```

Creating an entity via sensuctl with the --interactive flag now prompts for the entity name when not provided on the command line.
Check hooks with stdin: true now receive actual event data on STDIN instead of an empty event.

> 

Reviewer: ```suggestion
Some issues with check scheduling and updating have been fixed by refactoring the backend's watcher implementation.

```

**KNOWN ISSUES:**

> Deleting an asset via the API or sensuctl will only delete the asset resource within Sensu backend. 

Reviewer: I think this seems redundant with the next line so I suggest striking it

```suggestion
```

> Deleting an asset will not remove references to said asset. It is the operators responsibility to remove the asset from the runtime_assets field of the check, hook, filter, mutator, or handler.

Reviewer: ```suggestion
Authentication via OIDC is not yet supported in the Web UI.
Deleting an asset will not remove references to said asset. It is the operators responsibility to remove the asset from the runtime_assets field of the check, hook, filter, mutator, or handler.
```
